tests/e2e/restart_test.go

```diff
@@ -20,97 +18,141 @@
 package e2e
 
 import (
-	"log"
-	"os"
-
 	. "gopkg.in/check.v1"
 
 	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
 )
 
-type PluginSuite struct {
+type RestartTestData struct {
+	config        *inputparams.TestConfig
 	volumeName    string
-	hostIP        string
 	containerName string
-	esxIP         string
 }
 
-func (s *PluginSuite) SetUpTest(c *C) {
-	s.hostIP = os.Getenv("VM2")
-	s.esxIP = os.Getenv("ESX")
+func (s *RestartTestData) SetUpSuite(c *C) {
+	s.config = inputparams.GetTestConfig()
+	if s.config == nil {
+		c.Skip("Unable to retrieve test config, skipping docker/plugin restart tests")
+	}
+
 	s.volumeName = inputparams.GetUniqueVolumeName("restart_test")
 	s.containerName = inputparams.GetUniqueContainerName("restart_test")
 
+	// ensure there are no remanants from an earlier run
+	out, err := dockercli.RemoveContainer(s.config.DockerHosts[1], s.containerName)
+
 	// Create a volume
-	out, err := dockercli.CreateVolume(s.hostIP, s.volumeName)
+	out, err = dockercli.CreateVolume(s.config.DockerHosts[1], s.volumeName)
 	c.Assert(err, IsNil, Commentf(out))
 }
 
-func (s *PluginSuite) TearDownTest(c *C) {
```

Comment:
This is how the code was originally and wanted to allow docker create the volume when the container runs. Seems like AttachVolume() and AttachVolumeWithRestart() do not specify the volume driver name and hence while one test will use the vsphere volume the second one will use  local volume. Good. I'll fix.